src/github.com/matrix-org/dendrite/clientapi/routing/register.go

```diff
@@ -466,7 +483,7 @@ func handleRegistrationFlow(
 	case authtypes.LoginTypeDummy:
 		// there is nothing to do
 		// Add Dummy to the list of completed registration stages
-		sessions[sessionID] = append(sessions[sessionID], authtypes.LoginTypeDummy)
+		sessions.Set(sessionID, append(sessions.Get(sessionID), authtypes.LoginTypeDummy))
```

Comment:
Given this is the only place we use `Set`, I'd be tempted to instead have something like:

```golang
sessions.AddCompletedStage(sessionID, authtypes.LoginTypeDummy)
```